packages/quantic/cypress/e2e/default-2/recommendation-list/recommendation-list-expectations.ts

```diff
@@ -0,0 +1,126 @@
+      selector
+        .recommendations()
+        .should(display ? 'exist' : 'not.exist')
+        .logDetail(`${should(display)} display recommendations`);
+    },
+
+    recommendationsEqual: (recommendationsAlias: string) => {
+      cy.get<Array<{Title: string}>>(recommendationsAlias).then(
+        (recommendations) => {
+          selector
+            .recommendationLinks()
+            .then((elements) => {
+              return Cypress.$.makeArray(elements).map(
+                (element) => element.innerText
+              );
+            })
+            .should(
+              'deep.equal',
+              recommendations.map((result) => result.Title)
+            )
+            .logDetail('should render the received recommendations');
+        }
+      );
+    },
+
+    correctFieldsIncluded: (expectedFieldsToInclude: string[]) => {
+      cy.get<Interception>(InterceptAliases.Search)
+        .then((interception) => {
+          const fieldsToInclude = interception.request.body.fieldsToInclude;
+          expectedFieldsToInclude.forEach((field) =>
+            expect(fieldsToInclude).to.include(field)
+          );
+        })
+        .logDetail('should include the correct fields the request');
+    },
+
+    correctRecommendationId: (expectedRecommendationId: string) => {
+      cy.get<Interception>(InterceptAliases.Search)
+        .then((interception) => {
+          const recommendationId = interception.request.body.recommendation;
+          expect(recommendationId).to.equal(expectedRecommendationId);
+        })
+        .logDetail(
+          'should send the correct recommendation id value in the request'
+        );
+    },
+
+    correctNumberOfRecommendations: (
+      expectedNumberOfRecommendations: number
+    ) => {
+      cy.get<Interception>(InterceptAliases.Search)
+        .then((interception) => {
+          const recommendationId = interception.request.body.numberOfResults;
+          expect(recommendationId).to.equal(expectedNumberOfRecommendations);
+        })
+        .logDetail(
+          'should send the correct number of recommendations value in the request'
+        );
+    },
+
+    correctNumberOfRecommendationsPerRow: (
+      expectedNumberOfRecommendationsRow: number
+    ) => {
+      selector
+        .recommendations()
+        .invoke('css', 'width')
+        .then((width) => {
+          const recommendationWidth = parseInt(width.toString());
+          selector
+            .get()
+            .invoke('css', 'width')
+            .then((componentWidth) => {
+              const expectedRecommendationWidth =
+                parseInt(componentWidth.toString()) /
+                expectedNumberOfRecommendationsRow;
+              expect(recommendationWidth).to.equal(expectedRecommendationWidth);
+            });
+        })
+        .logDetail(
+          `should display ${expectedNumberOfRecommendationsRow} recommendations per row`
+        );
+    },
+
+    logRecommendationOpen: () => {
```

Comment:
I think this should be fleshed out no?

Just testing for the right usage analytics name is not really enough?

We should be checking that the payload is also correct? not entirely but some key fields at least

Reply:
There should also be a `logRecommendationSearch` IMO as a test

Reply:
payload check added 👌
`logRecommendationSearch` added as well as `logRecommendationInterfaceLoad`